### PR TITLE
Pal 666 service by name only

### PR DESCRIPTION
**Pull request by nw1984**

### CODE REVIEW

File: clients-simple/src/main/resources/application-eureka.yaml [hunk 1]
```diff
@@ -0,0 +1,19 @@
+  eureka:
```
Comment (ac74475) [inline clients-simple/src/main/resources/application-eureka.yaml:1]: Add Licence header

File: clients-simple/src/main/resources/application-k8s.yaml [hunk 2]
```diff
@@ -0,0 +1,4 @@
+web:
```
Comment (ac74475) [inline clients-simple/src/main/resources/application-k8s.yaml:1]: Add Licence header

File: clients-simple/src/main/resources/application-static.yaml [hunk 3]
```diff
@@ -0,0 +1,4 @@
+web:
```
Comment (ac74475) [inline clients-simple/src/main/resources/application-static.yaml:1]: Add Licence header

File: clients-simple/src/main/resources/application.yaml [hunk 4]
```diff
@@ -0,0 +1,5 @@
+
```
Comment (ac74475) [inline clients-simple/src/main/resources/application.yaml:1]: Add Licence header

File: clients-simple/src/main/resources/application.yaml [hunk 5]
```diff
@@ -0,0 +1,5 @@
...
+web:
+  client:
+    palisade-service: "palisade-service-not-configured"
+    data-service: "palisade-service-not-configured"
```
Comment (ac74475) [inline clients-simple/src/main/resources/application.yaml:5]:
```suggestion
    data-service: "data-service-not-configured"
```

File: clients-simple/src/main/java/uk/gov/gchq/palisade/clients/simpleclient/client/SimpleClient.java [hunk 6]
```diff
@@ -17,11 +17,9 @@
 package uk.gov.gchq.palisade.clients.simpleclient.client;
 
 import com.netflix.discovery.EurekaClient;
```
Comment (dev930018) [inline clients-simple/src/main/java/uk/gov/gchq/palisade/clients/simpleclient/client/SimpleClient.java:19]: Delete usages of EurekaClient, shouldn't be needed

File: clients-simple/src/main/java/uk/gov/gchq/palisade/clients/simpleclient/client/SimpleClient.java [hunk 7]
```diff
@@ -93,7 +89,8 @@ private DataRequestResponse makeRequest(final String fileName, final String reso
             readRequest.setOriginalRequestId(uuid);
 
             LOGGER.info("Resource {} has DATA-SERVICE connection detail {}", resource.getId(), connectionDetail);
-            InputStream responseStream = dataClient.readChunked(dataService, readRequest).body().asInputStream();
+            DataClient dataClient = Feign.builder().target(DataClient.class, dataService.toString());
```
Comment (dev930018) [inline clients-simple/src/main/java/uk/gov/gchq/palisade/clients/simpleclient/client/SimpleClient.java:92]: This should also include the url field if possible

File: clients-simple/src/main/java/uk/gov/gchq/palisade/clients/simpleclient/web/DataClient.java [hunk 8]
```diff
@@ -29,7 +29,7 @@
 public interface DataClient {
 
     @PostMapping(value = "/read/chunked", consumes = "application/json", produces = "application/octet-stream")
-    Response readChunked(final URI url, @RequestBody final ReadRequest request);
+    Response readChunked(@RequestBody final ReadRequest request);
 
     @PostMapping(value = "/addSerialiser", consumes = "application/json", produces = "application/json")
     Boolean addSerialiser(final URI url, @RequestBody final AddSerialiserRequest request);
```
Comment (dev930018) [inline clients-simple/src/main/java/uk/gov/gchq/palisade/clients/simpleclient/web/DataClient.java:35]: Remove this usage of URL.
In fact, could remove method altogether.

File: clients-simple/src/main/java/uk/gov/gchq/palisade/clients/simpleclient/web/DataClient.java [hunk 9]
```diff
@@ -29,7 +29,7 @@
 public interface DataClient {
```
Comment (dev930018) [inline clients-simple/src/main/java/uk/gov/gchq/palisade/clients/simpleclient/web/DataClient.java:29]: Remove abofe @FeignClient annotation

File: clients-simple/src/main/resources/application-eureka.yaml [hunk 10]
```diff
@@ -0,0 +1,33 @@
...
+      serviceUrl:
+        defaultZone: http://${eureka.instance.hostname}:${eureka.instance.port}/eureka/
+    enabled: true
+    registerWithEureka: false
+    fetchRegistry: true
+    registry-fetch-interval-seconds: 5
+  web:
+    client:
+      palisade-service: palisade-service
+      data-service: data-service
```
Comment (dev930018) [inline clients-simple/src/main/resources/application-eureka.yaml:33]: After adding url=${web.client.xxx-service} to the Feign builder, these should be ""

File: clients-simple/src/main/resources/application.yaml [hunk 11]
```diff
@@ -0,0 +1,18 @@
...
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+web:
+  client:
+    palisade-service: "palisade-service-not-configured"
+    data-service: "data-service-not-configured"
```
Comment (dev930018) [inline clients-simple/src/main/resources/application.yaml:18]: @dev-958 's preference seemed to be to use the static config here

Comment (dev-958) [reply clients-simple/src/main/resources/application.yaml:18]: yes please

**Pull request closed without merging**